core/src/test/java/org/hibernate/ogm/test/jpa/JPAStandaloneNoOGMTest.java

```diff
@@ -39,7 +39,7 @@ public void testJTAStandaloneNoOgm() throws Exception {
 			fail( "Expected exception was not raised" );
 		}
 		catch ( PersistenceException pe ) {
-			assertThat( Throwables.getRootCause( pe ).getMessage() ).contains( "hibernate.dialect" );
+			assertThat( Throwables.getRootCause( pe ).getMessage() ).contains( "No Persistence provider for EntityManager named noogm" );
```

Comment:
Wondering if the purpose of the test was not also to validate that the error message was explicit enought about the issue (i.e. the fact that `hibernate.dialect` was not defined). Maybe the message is in the `cause`?

Reply:
I'm pretty sure the exception thrown was completely different but I will double check

Reply:
OK, that would be a bit unfortunate. Maybe it's worth digging to see if ORM forgot to add a cause or if it's totally impossible to have a precise error message.

Reply:
This change in behavior is due to the changes in `HibernatePersistenceProvider.createEntityManagerFactory`.

Per JPA specification this method doesn't have to throw exceptions. Meaning that, instead of throwing the exception with error message will return `org.hibernate.HibernateException: Access to DialectResolutionInfo cannot be null when 'hibernate.dialect' not set` it will return null

For this reason our test fails, because a different exception is thrown but the reason is the same.
I think this new exception explains bettere what happened then the old one. In fact in the cofiguration we left the persistence provider away on purpose.

I guess now we will have to update the OGM provider  to not throw exceptions anymore as well

Reply:
OK, thanks for taking the time to explore this. Yes, we probably need to be in par with what is done in ORM. I have seen that the behavior might change depending on the compliance setting so, if it's indeed the case, we need to have a similar behavior.

Reply:
> Per JPA specification this method doesn't have to throw exceptions. 

Interesting, I'm familiar with that bootstrap process and Steve is right (of course?) on that HHH JIRA on which we were discussing this, but this doesn't look like necessary the case covered by the specification.

Not a blocker for OGM but I'll go see some clarifications.